stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java

```diff
@@ -61,8 +62,18 @@ public static Object nativeRespond(ObjectValue connectionObj, ObjectValue outbou
         HttpCarbonMessage inboundRequestMsg = HttpUtil.getCarbonMsg(connectionObj, null);
         Strand strand = Scheduler.getStrand();
         DataContext dataContext = new DataContext(strand, new NonBlockingCallback(strand), inboundRequestMsg);
-        HttpCarbonMessage outboundResponseMsg = HttpUtil
-                .getCarbonMsg(outboundResponseObj, HttpUtil.createHttpCarbonMessage(false));
+        HttpCarbonMessage outboundResponseMsg;
+        if (outboundResponseObj.get(RESPONSE_CACHE_CONTROL_FIELD) == null && outboundResponseObj.
+                getNativeData(HttpConstants.IS_RESPONSE_PROCESSED) != null) {
+            String errorMessage = "Couldn't complete the outbound response which has been already processed.";
```

Comment:
`already processed` wording does not specifically say that the response is reused. Shall we rephrase it?
```suggestion
            String errorMessage = "Couldn't complete the respond operation as the response has been already used.";
```